internal/metrics/metrics.go

```diff
@@ -0,0 +1,68 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// When adding metric names, see https://prometheus.io/docs/practices/naming/#metric-names
+const (
+	edgeDeviceSuccessfulRegistration = "k4e_operator_edge_devices_successful_registration"
+	edgeDeviceFailedRegistration     = "k4e_operator_edge_devices_failed_registration"
+	createdOBCQuery                  = "k4e_operator_created_obc"
```

Comment:
What do you think about adding metrics related to `edgeDeployment`?

Reply:
I added and removed it because @masayag and I weren't sure how it can help us for the test scaling... 
@masayag

Reply:
I miss to see how the notation of `edgedeployment` on itself is useful for us.
Theoretically, we can have tons of edgedeployments, without an edge device to be played on.
Therefore, when detached from the context of edge-device, the edge-deployments won't provide useful information.

However, they are meaningful in the context of edge-device. We can monitor how many edge-deployments are successfully running (or failed to run), to create an alert based on specific rule (e.g. if failed to run a specific edgedeployment on % of the devices might indicate on an issue with the deployment itself).
however, the latter requires a design to define the requirements for it (for observability purpose).